.ci/deploy.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+set -eo pipefail # Exit with nonzero exit code if anything fails
+
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Pull requests and commits to other branches shouldn't try to deploy, just build to verify
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+    echo "Skipping deploy."
+    exit 0
+fi
+
+if [ -z "$ENCRYPTION_LABEL" ]; then
+    echo "No ENCRYPTION_LABEL set; Falling back to Travis deployment."
+    exit 0
+fi
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deploy)
+git clone $REPO temp_out
+cd temp_out
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+mv .git ../out/.git
+cd ../out
+
+# Now let's go have some fun with the cloned repo
+git config user.name "Travis CI"
+git config user.email "$COMMIT_AUTHOR_EMAIL"
+
+# If there are no real changes to the compiled out
+# e.g. this is a README update or build time update in html
+# then just bail.
+if git diff --quiet -- . ':(exclude)index.html'; then
```

Comment:
I'm still thinking about this, what if it changes the layout ..

Reply:
nah, dont bother with this exclude... just push it all.
The only improvement is keeping the `gh-pages` filestamps stable, so the json's last modification can be deduced

Reply:
(approving; up to you what you do with this...)

Reply:
Yeah, I'll remove the exclude, it's overcomplicating.